db/wal_manager.cc

```diff
@@ -492,7 +492,9 @@ Status WalManager::ReadFirstLine(const std::string& fname,
                           Status::Corruption("log record too small"));
       // TODO read record's till the first no corrupt entry?
     } else {
-      WriteBatch batch;
+      // We create a new batch and make sure it has a valid prot_info_ to store
+      // the data checksums.
+      WriteBatch batch(0, 0, 8, 0);
```

Comment:
I think this the SetContents() used during recovery is this one: https://github.com/facebook/rocksdb/blob/5a085d789d0ec5c896acfd1560bb0701c5de6a42/db/db_impl/db_impl_open.cc#L964

The one here doesn't write the key/values to anywhere so doesn't need integrity protection.